protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -109,13 +92,37 @@ func (rpccs *RPCConsumerServer) ServeRPCRequests(ctx context.Context, listenEndp
 
 	go chainListener.Serve(ctx, cmdFlags)
 
+	initialRelays := true
+	rpccs.relaysMonitor = relaysMonitor
+
 	// we trigger a latest block call to get some more information on our providers, using the relays monitor
-	if RelaysHealthEnable {
+	if cmdFlags.RelaysHealthEnableFlag {
+		rpccs.relaysMonitor.SetRelaySender(func() (bool, error) {
+			success, err := rpccs.sendCraftedRelaysWrapper(initialRelays)
+			if success {
+				initialRelays = false
+			}
+			return success, err
+		})
 		rpccs.relaysMonitor.Start(ctx)
+	} else {
+		rpccs.sendCraftedRelaysWrapper(true)
 	}
 	return nil
 }
 
+func (rpccs *RPCConsumerServer) sendCraftedRelaysWrapper(initialRelays bool) (bool, error) {
+	if initialRelays {
+		// Only start after everything is initialized - check consumer session manager
+		ok, err := rpccs.waitForPairing()
+		if !ok {
```

Comment:
what if err != nil but ok == true? lets minimize the waitForPairing to err or not

Reply:
Fixed in 2de5082ec9b78d8ecd5a34d9994ef8988fbcf5c8